Cython/Compiler/Tests/TestTreeFragment.py

```diff
@@ -59,6 +61,39 @@ def test_temps(self):
         self.assertTrue(isinstance(s[1].rhs, TempRefNode))
         self.assertTrue(s[0].expr.handle is s[1].rhs.handle)
 
+    def test_parse_fault_tolerant_complete(self):
+        code = u'''
+def method():
+    pass
+
+def method2():
+    error_here =
+
+def method3():
+    error_here.
+
+def method4():
+    without_error = 10
+
+def method5():
+    error_here =
+'''
+        with local_errors() as errors:
+            stats = parse_from_strings("test_name", code, fault_tolerant=True).body.stats
+        self.assertIsInstance(stats[0], DefNode)
+        self.assertIsInstance(stats[0].body, PassStatNode)
+        self.assertIsInstance(stats[1], DefNode)
+        self.assertIsNone(stats[1].body)
+        self.assertIsInstance(stats[2], DefNode)
+        self.assertIsInstance(stats[2].body, ExprStatNode)
+        self.assertIsInstance(stats[3], DefNode)
+        self.assertIsInstance(stats[3].body, SingleAssignmentNode)
+        self.assertIsInstance(stats[4], DefNode)
```

Comment:
It'd be nice that you can check the node name too to know which method was recognized.

Reply:
Also, please check `len(stats)` to know how many statements were generated.